### PR TITLE
DAOS-15064 mgmt: Show all ranks in daos system query

### PR DESCRIPTION
**Pull request by kjacque**
A bug was limiting the number of ranks shown to the number of MS replicas instead.

- Show all rank fabric URIs.
- Added MS ranks list to sys info.
- Add MS ranks to daos system query output.

Features: control

Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/mgmt/cli_mgmt.c [hunk 1]
```diff
@@ -475,21 +475,29 @@ dc_mgmt_get_sys_info(const char *sys, struct daos_sys_info **out)
 	if (info == NULL)
 		D_GOTO(out_attach_info, rc = -DER_NOMEM);
 
+	D_ALLOC_ARRAY(info->dsi_ms_ranks, resp->n_ms_ranks);
+	if (info->dsi_ms_ranks == NULL)
+		D_GOTO(err_info, rc = -DER_NOMEM);
+	memcpy(info->dsi_ms_ranks, resp->ms_ranks, resp->n_ms_ranks);
```
Comment (knard-intel) [inline src/mgmt/cli_mgmt.c:481]:
```suggestion
	memcpy(info->dsi_ms_ranks, resp->ms_ranks, resp->n_ms_ranks * sizeof(uint32_t));
```

**Review by daltonbohning — COMMENTED** [review]
ftest LGTM

File: src/mgmt/cli_mgmt.c (hunk 1 above)
Comment (kjacque) [reply src/mgmt/cli_mgmt.c:481]: Oops! Good catch.

**Review by mjmac — APPROVED** [review]
LGTM, thanks for adding the access_point_ranks output. Just FYI, I'm planning to move this system query functionality into a shared method in #14730. I'll update that one after this work lands.

**Review by knard-intel — APPROVED** [review]

**Review by tanabarr — APPROVED** [review]

**Pull request merged**